src/YesSql.Core/Store.cs

```diff
@@ -117,21 +117,53 @@ public async Task InitializeCollectionAsync(string collectionName)
 
-                        selectCommand.CommandText = $"SELECT 1 FROM {Dialect.QuoteForTableName(Configuration.TablePrefix + documentTable)}";
+                        var selectBuilder = Dialect.CreateBuilder(Configuration.TablePrefix);
+                        selectBuilder.Select();
+                        selectBuilder.AddSelector("*");
+                        selectBuilder.From(Configuration.TablePrefix + documentTable);
+                        selectBuilder.Take("1");
+
+                        selectCommand.CommandText = selectBuilder.ToSqlString();
                         selectCommand.Transaction = transaction;
                         Configuration.Logger.LogTrace(selectCommand.CommandText);
-                        var result = await selectCommand.ExecuteScalarAsync();
-
-                        transaction.Commit();
-                        
-                        // Table already exists?
-                        if (result != null && Convert.ToInt64(result) == 1)
+                        using (var result = await selectCommand.ExecuteReaderAsync())
                         {
-                            return;
+                            transaction.Commit();
+
+                            if (result != null)
+                            {
+                                try
+                                {
+                                    // Check if the Version column exists
+                                    result.GetOrdinal(nameof(Document.Version));
+                                }
+                                catch
+                                {
+                                    using (var migrationTransaction = connection.BeginTransaction())
+                                    {
+                                        var migrationBuilder = new SchemaBuilder(Configuration, migrationTransaction);
+
+                                        try
+                                        {
+                                            migrationBuilder
+                                                .AlterTable(documentTable, table => table
+                                                    .AddColumn<long>(nameof(Document.Version), column => column.WithDefault(0))
+                                                );
+
+                                            migrationTransaction.Commit();
+                                        }
+                                        catch
+                                        {
+                                            // Another thread must have altered it
+                                        }
+                                    }
+                                }
+                                return;
+                            }
                         }
                     }
                 }
-                catch
+                catch 
                 {
                     using (var transaction = connection.BeginTransaction())
```

Comment:
Remove this inner catch

Reply:
I think it needs it to handle the table create if the table itself doesn't exist?

Reply:
right sorry, I thought it was the one I added, but you already removed it.